Mathlib/Data/Nat/Cast/Defs.lean

```diff
@@ -47,11 +47,17 @@ class Nat.AtLeastTwo (n : ℕ) : Prop where
 instance instNatAtLeastTwo : Nat.AtLeastTwo (n + 2) where
   prop := Nat.succ_le_succ $ Nat.succ_le_succ $ Nat.zero_le _
 
-lemma Nat.AtLeastTwo.ne_zero (n : ℕ) [h : n.AtLeastTwo] : n ≠ 0 := by
-  rintro rfl; exact absurd h.1 (by decide)
+namespace Nat.AtLeastTwo
 
-lemma Nat.AtLeastTwo.ne_one (n : ℕ) [h : n.AtLeastTwo] : n ≠ 1 := by
-  rintro rfl; exact absurd h.1 (by decide)
+variable {n : ℕ} [n.AtLeastTwo]
+
+lemma one_lt : 1 < n := prop
+lemma ne_one : n ≠ 1 := Nat.ne_of_gt one_lt
+lemma one_le : 1 ≤ n := Nat.le_of_lt one_lt
+lemma pos : 0 < n := one_le
+lemma ne_zero : n ≠ 0 := Nat.ne_of_gt pos
```

Comment:
I don't think there is such an instance. I'll investigate adding that and removing these lemmas.

Reply:
Adding `import Mathlib.Algebra.NeZero` triggered this `assert_not_exists`. What's the right way to address this error?

https://github.com/leanprover-community/mathlib4/blob/4acfd8a235ca7c92ee2dc6d96d12a96aafcbddc7/Mathlib/Algebra/Field/Defs.lean#L178-L181

Also, `NeZero.one_le` didn't exist, so I had to add it. The code where these `NeZero` lemmas are used seems a bit less uniform now with each lemma looking a bit different to use:
- `(NeZero.ne n)` and `(NeZero.pos n)`
- `NeZero.one_le`
- `Nat.AtLeastTwo.ne_one` and `Nat.AtLeastTwo.one_lt`
- `Nat.AtLeastTwo.prop`

Would it be worth keeping these lemmas all under `Nat.AtLeastTwo` for uniformity? Maybe it doesn't matter since `exact?` can find all of them..

Reply:
I solved the import error by moving this instance to a later file